drivers/memc/memc_stm32_sdram.c

```diff
@@ -129,5 +129,7 @@ static const struct memc_stm32_sdram_config config = {
 	.banks_len = ARRAY_SIZE(bank_config),
 };
 
+#if DT_NODE_HAS_STATUS(DT_INST(0, st_stm32_fmc_sdram), okay)
```

Comment:
This is not required

Reply:
Why is it that if I set nodes to "disabled" then ram is still initialized? It doesn't seem to work unless I add this? But maybe there is some better way. Can you suggest a better solution?

Reply:
Kconfig for SDRAM is selected based on DT status, see https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/memc/Kconfig.stm32#L15. If Kconfig setting is not set, this file will not be compiled https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/memc/CMakeLists.txt#L3

Reply:
What about if I enable it in my _defconfig? I think it should work regardless of how the setting is set. I actually had to explicitly trace the code in the debugger to figure out why my memory controller was still enabled even though I disabled it in the device tree. So actually it should be possible to disable the node in the device tree only and then it should not be configured in code. For example, if you want to temporarily disable the memory controller and use the pins with some other configuration. Then you would have to disable it in menuconfig but it would be easier to just set the node to status "disabled". Am I missing something here?

Reply:
If you don't need memory controller at all just make sure `CONFIG_MEMC=n` (default) and nothing will be initialized.

Reply:
Is it not the case that all drivers should honor "status" field? Ie if status = "disabled" then it should not configure a bunch of things that are expected to be disabled? @MaureenHelm can you give us some input on this?

Reply:
Honestly, for me, it took me by surprise when I set it to disabled and memory still continued to work. It seemed wrong so that's why I added this commit..

Reply:
@gmarull I notice you are using #if DT_NODE_HAS_STATUS(DT_NODELABEL(sdram2), okay) in the linker script. I think we should be using the same thing in the driver C code but for the memc as well. Otherwise we can end up with disabled nodes that are still initialized which would be kindof weird.

Reply:
In this case it is clearly not necessary as Kconfig entry takes into account DT status, so the file will not get compiled if not needed.

Reply:
It is still a problem if you enable it in kconfig but your devicetree disables it. If it is disabled in device tree it should rely on compiler to remove dead code by not instantiating it. So if this is not done in otherdrivers then it is an error.